### PR TITLE
[12.0] Backport/forward port mass editing improvements from 13.0/10.0

### PR DESCRIPTION
**Pull request by simahawk**
From v13: refactor based on mass_operation_abstract https://github.com/OCA/server-ux/commit/8b7156422f8004d15bbdf53257ef8989c275ad4e

From this PR from v10 https://github.com/OCA/server-tools/pull/1856

* improve mig scripts
* ease override of field options

Preserved (although rewritten) improvement from v12 for security group https://github.com/OCA/server-ux/commit/620a58e5e57f6df3dff4cc0bcaa848b34a0393a9
This one should be fwd ported to v13 once this is over.

### CODE REVIEW

**Review by lmignon — COMMENTED** [review]
LGTM (Code review + functionnal tests)
Unfortunately, it seems that something is wrong into the xml files (runbot is red) 
Here it's the warnings into runbot log
``` 
2020-10-05 09:03:32,289 159 INFO openerp_test odoo.modules.loading: Module mass_editing: loading demo 
2020-10-05 09:03:32,369 159 WARNING openerp_test odoo.models: ir.actions.act_window.create() with unknown fields: group_ids
2020-10-05 09:03:32,374 159 WARNING openerp_test odoo.models: ir.actions.act_window.create() with unknown fields: group_ids
```

**Comment by simahawk** [comment]
@lmignon runbot :green_circle: as well

**Review by lmignon — APPROVED** [review]

**Comment by lmignon** [comment]
ping @Yajo

**Review by HaraldPanten — APPROVED** [review]

File: mass_operation_abstract/views/view_mass_operation_wizard_mixin.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+</odoo>
+
+
+<!--
+
+
+
+
+ -->
```
Comment (yajo) [inline mass_operation_abstract/views/view_mass_operation_wizard_mixin.xml]: :thinking: :question:

File: requirements.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 pyzbar
 pdf2image
+openupgradelib @ git+https://github.com/OCA/openupgradelib.git
```
Comment (yajo) [inline requirements.txt:3]: I don't think this format is correct... At least I never saw it. I think it's just:

```suggestion
git+https://github.com/OCA/openupgradelib.git
```

**Review by yajo — APPROVED** [review]
Seems good.

File: requirements.txt (hunk 2 above)
Comment (lmignon) [reply requirements.txt:3]: @Yajo This format is the right one to declare a python dependency installed from a git repository into a requirements.txt file

File: mass_operation_abstract/views/view_mass_operation_wizard_mixin.xml (hunk 1 above)
Comment (lmignon) [reply mass_operation_abstract/views/view_mass_operation_wizard_mixin.xml]: @simahawk Can you remove this comment so we can move forward?

File: requirements.txt (hunk 2 above)
Comment (yajo) [reply requirements.txt:3]: OK, TIL. :smile:

File: mass_operation_abstract/views/view_mass_operation_wizard_mixin.xml (hunk 1 above)
Comment (simahawk) [reply mass_operation_abstract/views/view_mass_operation_wizard_mixin.xml]: WTH does it come from?? :thinking: :rofl: 
Done :wink:

**Comment by lmignon** [comment]
@simahawk Is this PR deserve a new major version number? IMO yes

**Comment by simahawk** [comment]
it has "12.0.2.0.0" already -> the upgrade script is linked to it

**Comment by lmignon** [comment]
@simahawk I propose to merge with `oca merge patch` otherwise the version will  no be available on pypi...  (it will be available as pre release....)

**Comment by simahawk** [comment]
@lmignon then go for minor as some improvements has been added anyway

**Pull request merged**